app/Http/Controllers/GazetteerController.php

```diff
@@ -309,7 +311,7 @@ private function searchDataitems($parameters)
         }
 
         /* BUILD SEARCH QUERY WITH PARAMS */
-        if ($parameters['recordtypeid']) $dataitems->where('recordtype_id', '=', $parameters['recordtypeid']);
+        if ($parameters['recordtype']) $dataitems->where('tlcmap.recordtype' . '.type', '=', $parameters['recordtype']);  // Filter by recordtype value
         if ($parameters['lga']) $dataitems->where('lga', '=', $parameters['lga']);

```

Comment:
Continuing the above comment, a cleaner solution would be just adding a condition on the recordtype relationship.

Try a solution similar to this (Remove the join part from line 232).

```php
$dataitems->whereHas('recordtype', function (Builder $query) use ($parameters) {
    $query->where('type', 'id', $parameters['recordtype']);
});
```